lib/ansible/galaxy/collection.py

```diff
@@ -310,6 +310,14 @@ def from_name(collection, apis, requirement, force, parent=None):
 
         for api in apis:
             collection_url_paths = [api.api_server, 'api', 'v2', 'collections', namespace, name, 'versions']
+
+            available_api_versions = get_available_api_versions(api)
+            if 'v3' in available_api_versions:
+                # /api/v3/ exists, use it
+                collection_url_paths[2] = 'v3'
```

Comment:
It sounds like this should be set on the api object and used as part of the original url paths